nntrainer/optimizers/optimizer_internal.h

```diff
@@ -151,23 +136,12 @@ class Optimizer : public ml::train::Optimizer {
   virtual void setProperty(const PropertyType type,
                            const std::string &value = "");
 
-  /**
-   * @brief     get the base name for the optimizer
-   * @retval    base name of the optimizer
-   */
-  virtual std::string getBaseName() = 0;
```

Comment:
this is still needed, mutliple optimizers with same name can exist and might be added to the model later.
Each optimizer must be able to provide a basename in order to build it a unique name.

Reply:
I think `getBaseName` is exactly same as `getType`. Would there be a reason to distinguish those two?

Reply:
I thought this would fail as getBaseName was being used to assign names. But its not being used for optimizer.
getType() works as long as it is unique.